ostree_upload_server/importers/tar.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import tempfile
+import tarfile
+
+import urllib
+import urlparse
+
+from gi.repository import GLib
+
+from os import path
+from shutil import rmtree
+
+from .base import BaseImporter
+from .util import find_repo, open_repository
+
+class TarImporter(BaseImporter):
+    MIME_TYPE = 'application/x-tar'
+
+    def _import_commit(self, target_repo, src_path_obj, refs=None):
+        if not self._source_repo_path:
+            raise RuntimeError("Cannot invoke _import_commit without calling import_to_repo first")
+
+        if not refs:
+            raise RuntimeError("Cannot invoke _import_commit without valid refs")
+
+        logging.debug("Importing %s%s into %s...", self._source_repo_path,
+                      refs,
+                      target_repo.get_path().get_uri())
+
+        options = GLib.Variant('a{sv}', {
+            'refs': GLib.Variant('as', refs),
+            'inherit-transaction': GLib.Variant('b', True),
+        })
+
+        source_repo_uri = urlparse.urljoin('file:', urllib.pathname2url(self._source_repo_path))
```

Comment:
You can just get this from the source repo - `repo.get_path().get_uri()` will get the URI from `Gio.File`.

Reply:
Changing it that way would keep the repo object around longer than it needs to be instead of just using the simple path string but I can change it if you want.

Reply:
Alright.